File: algorithms/12_prime_number.c
#include <stdio.h>

int main()
{
    int number;

    printf("Digite um número: \n");
    scanf("%d", &number);

    if (number == 1 || number == 0)
    {
        printf("Não é um número primo!\n");
        return 0;
    }
    else if (number == 2)
    {

        printf("É um número primo!");
        return 0;
    }
    else
    {
        if (number % 2 != 0)
        {
            for (int i = 3; i < number; i += 2)
            {
                if (number % i == 0)
                {
                    printf("Não é um número primo!\n");
                    return 0;
                }
            }

            printf("É um número primo!");
        }
        else
        {
            printf("Não é um número primo!\n");
        }
    }

    return 0;
}

File: algorithms/12_prime_number.py
number = int(input("Digite um número: "))

if number == 1 or number == 0:
    print("Não é um número primo!")
elif number == 2:
    print("É um número primo!")
else:
    if number % 2 != 0:
        i = 3
        while i < number:
            if number % i == 0:
                print("Não é um número primo!")
                exit()

            i += 2
        print("É um número primo!")
    else:
        print("Não é um número primo!")
